server/src/main/java/org/opensearch/extensions/action/ExtensionTransportAction.java

```diff
@@ -10,44 +10,34 @@
 
 import org.opensearch.action.ActionListener;
 import org.opensearch.action.support.ActionFilters;
-import org.opensearch.action.support.HandledTransportAction;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.inject.Inject;
-import org.opensearch.common.settings.Settings;
+import org.opensearch.action.support.TransportAction;
 import org.opensearch.extensions.ExtensionsManager;
-import org.opensearch.node.Node;
 import org.opensearch.tasks.Task;
-import org.opensearch.transport.TransportService;
+import org.opensearch.tasks.TaskManager;
 
 /**
- * The main proxy transport action used to proxy a transport request from extension to another extension
+ * A proxy transport action used to proxy a transport request from an extension to execute on another extension
  *
  * @opensearch.internal
  */
-public class ExtensionTransportAction extends HandledTransportAction<ExtensionActionRequest, ExtensionActionResponse> {
+public class ExtensionTransportAction extends TransportAction<ExtensionActionRequest, RemoteExtensionActionResponse> {
 
-    private final String nodeName;
-    private final ClusterService clusterService;
     private final ExtensionsManager extensionsManager;
 
-    @Inject
     public ExtensionTransportAction(
-        Settings settings,
-        TransportService transportService,
+        String actionName,
         ActionFilters actionFilters,
-        ClusterService clusterService,
+        TaskManager taskManager,
```

Comment:
I am missing where actionFilters and taskManager are being used.

Reply:
The `TransportAction` superclass: `super(actionName, actionFilters, taskManager);`

The taskManager maintains the thread context and the action filters are chained during execution of the action.